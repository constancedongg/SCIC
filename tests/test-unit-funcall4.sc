|'{km} = 0.001 '{m}|;
|'{ms} = 1000.0 '{s}|;

/* return value will be auto-converted */
float '{m/s} func foo() {
    float '{cm} x = 250.0;
    float '{s} t = 2.5;
    return x / t;
}

float '{km/s} func bar(float '{cm/ms} a) {
    float '{cm/ms} b = 31.2 ^ 2;
    printf(a);
    printf(b);
    if (a > b) {
        return a + b;
    }
    else {
        return b - a;
    }
}

int func main() {
    printf(foo()); 
    printf(bar(foo()));
    return 0;
}