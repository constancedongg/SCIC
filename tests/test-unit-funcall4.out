1
0.1
973.44
9.7334
